src/main/java/org/jitsi/jigasi/transcription/TranslationManager.java

```diff
@@ -0,0 +1,162 @@
+
+    /**
+     * The translationService to be used for translations.
+     */
+    private final TranslationService translationService;
+
+    /**
+     * Initializes the translationManager with a TranslationService
+     * and adds the default target language to the list.
+     *
+     * @param translationService to be used by the TranslationManger
+     */
+    public TranslationManager(TranslationService translationService)
+    {
+        this.translationService = translationService;
+        String defaultTargetLanguage = getTargetLanguage();
+
+        if(!defaultTargetLanguage.isEmpty())
+        {
+            addTargetLanguage(defaultTargetLanguage);
+        }
+    }
+
+    /**
+     * Adds a {@link TranslationResultListener} to the list of
+     * listeners to be notified of a TranslationResult.
+     *
+     * @param listener to be added.
+     */
+    public void addListener(TranslationResultListener listener)
+    {
+        this.listeners.add(listener);
+    }
+
+    /**
+     * Adds a language tag to the list of target languages for
+     * translation.
+     *
+     * @param language to be added to the list
+     */
+    public void addTargetLanguage(String language)
+    {
+        this.languages.add(language);
+    }
+
+    /**
+     * Translates the received {@link TranscriptionResult} into required languages
+     * and returns a list of {@link TranslationResult}s.
+     *
+     * @param result the TranscriptionResult notified to the TranslationManager
+     * @return list of TranslationResults
+     */
+    public ArrayList<TranslationResult> getTranslations(TranscriptionResult result)
+    {
+        ArrayList<TranslationResult> translatedResults
+            = new ArrayList<>();
+        for(String language : languages)
+        {
+            String translatedText = translationService.translate(
+                result.getAlternatives().iterator().next().getTranscription(),
+                result.getLanguage().substring(0,2),
+                language);
+
+            translatedResults.add(new TranslationResult(
+                result,
+                language,
+                translatedText));
+        }
+        return translatedResults;
+    }
+
+    /**
+     * Translates the received {@link TranscriptionResult} into the
+     * target languages and notifies the {@link TranslationResultListener}'s
+     * of the {@link TranslationResult}s.
+     *
+     * @param result the result which has come in.
+     */
+    @Override
+    public void notify(TranscriptionResult result)
+    {
+        if(!result.isInterim())
+        {
+            ArrayList<TranslationResult> translations
+                = this.getTranslations(result);
+            for (TranslationResultListener listener : listeners)
+            {
+                for (TranslationResult translation : translations)
+                    listener.notify(translation);
+            }
+        }
+    }
+
+    @Override
+    public void completed() {
+
```

Comment:
Do you think anything needs to be done here?

Reply:
the ArrayList of languages to translated could be cleared.